### DISCUSSION TITLE
How to Utilize Term.jl in Packages or Projects?

### DISCUSSION THREAD

**Discussion by TheCedarPrince** [FedeClaudi/Term.jl#50] [Q&A]
Hi @FedeClaudi ,

Congrats on the first release and great work here!
I was wondering, in terms of incorporating Term.jl into packages or projects, what is your opinion on how to do that?
For example, a majority of my work projects involve reading data (be it from a database or large data file like a CSV or binary format), processing or mutating that data within a dataframe, generating summary statistics off the data through a variety of means, and then generating reports.
To me, it seems like Term.jl would be super handy in my terminal-based workflow but I am just not sure where it could be incorporated. 

Perhaps another way of asking this question is do you have any ideas on where you think Term.jl is best suited to be used or have an idea of a user who could really benefit from this tool?
Sorry for the vagueness of this question - I see the potential of Term.jl but am struggling to find where I could incorporate it into my own workflows so just figured I'd ask if you had any ideas.

Thanks! 

~ tcp 🌳

**Comment by FedeClaudi** [comment]
Nice question!


I'd say that the starting point is to look at where you already have terminal output in your projects:
1. do you have progress bars showing things like how many files where loaded, how much of the data has been processed etc... maybe use Term's progress bars if they look nicer/give more info
2. are you printing messages or using logging macros to update the user while stuff happens? Use Term's logging styling or `tprint` strings with markup style info to highlight relevant bits of information and make it easier to parse

E.g. this is from a (work) project of mine
![image](https://user-images.githubusercontent.com/17436313/159301650-7724113d-39d1-4a92-a5d2-f29459d36810.png)

A week ago I was printing the same info to the screen but it was plain white text, this looks more interesting and its easier to parse (for me). 

Another place where it might be relevant is the summary statistics: you could make a `Panel` and have styled text inside to make it easier to parse. Or have multiple panels with different bits of info (e.g. number of file parsed, which processing steps where applied, summary statistics...) and create a layout to give all the info in an appealing format. 

A thing that might be cool too (and I want to implement this but didn't get around to it yet) is using `Tree` to show the files/folder structure of a directory. If it's relevant for you that might be nice. 


I guess beyond that it it's hard to give specific advice without knowing more about what you do. I hope this gets you started, but if not do let me know and we can chat about it.
